src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -22,9 +22,9 @@ namespace System
 {
     public static partial class Math
     {
-        public const double E = 2.7182818284590452354;
+        public const double E = 2.7182818284590451;
 
-        public const double PI = 3.14159265358979323846;
```

Comment:
Changing these to the rountripable value here does not look like an improvement to me.

`3.14159265358979323846...` is the correct value of PI. It is easy to see that it is correct. Yes, compiler rounds it up but that is fine.

Reply:
The translation of source code constant representation to binary is a lossy operation. For example, if you see `2A` byte in the binary, you cannot tell whether the source code said `0x2A`, `42`, `'*'` or infinite number of other variants. Human readable code should use the most suitable variant to maximize readability and to make it easy to reason about the code correctness. Floating point constants should be no exception from this rule.